WebView/app/src/main/java/com/android/samples/webviewdemo/MainActivity.kt

```diff
@@ -96,22 +96,22 @@ class MainActivity : AppCompatActivity() {
                         isMainFrame: Boolean,
                         replyProxy: JavaScriptReplyProxy
                     ) {
-                        onMessageReceived(message.data!!)
+                        invokeShareIntent(message.data!!)
                     }
                 })
         } else {
             webview.addJavascriptInterface(object {
                 @JavascriptInterface
                 fun postMessage(message: String) {
                     // Use the handler to invoke method on UI thread
-                    handler.post(Runnable { onMessageReceived(message) })
+                    handler.post(Runnable { invokeShareIntent(message) })
```

Comment:
Sorry, this may not have been clear. This function parameter (the function's callback) should be named `onMessageReceived`, the argument you pass (the implemented callback which does what you want) in should be named `invokeShareIntent`. Something like:

```kotlin
private fun createJsObject(..., onMessageReceived) {
  // createJsObject doesn't need to know onMessageReceived will invoke an
  // Intent, it just knows it's a callback it should invoke.
  // ...
  onMessageReceived(message);
}

private fun invokeShareIntent(...) {
  // invokeShareIntent() doesn't need to know this will be wired up to a JS
  // object, it just knows how to invoke a shareIntent.
  // ...
}
```